### PR TITLE
updata testbed playbook for ubuntu 17.04

### PR DESCRIPTION
**Pull request by lguohan**
TODO:
* add docker deb gpg key
* auto-generate p4p1 interface configuration

### CODE REVIEW

File: ansible/ansible.cfg [hunk 1]
```diff
@@ -155,7 +155,9 @@ callback_whitelist = profile_tasks
 # wanting to use, for example, IP information from one group of servers
 # without having to talk to them in the same playbook run to get their
 # current IP information.
-fact_caching = memory
+fact_caching = jsonfile
+fact_caching_connection = ~/.ansible/cache
+fact_caching_timeout = 86400
```
Comment (pavel-shirshov) [inline ansible/ansible.cfg]: I think it's too long to cache all facts to one day. What if some interface was up and now it's down?
We should be able to track such things from our playbook.

File: ansible/roles/vm_set/tasks/main.yml [hunk 2]
```diff
@@ -13,8 +13,47 @@
...
 
+- name: Check if kernel upgrade needed
+  set_fact:
+    kernel_upgrade_needed: true
+  when:
+    - ansible_distribution == "Ubuntu"
+    - ansible_distribution_version == "17.04"
+    - ansible_kernel.find('4.10.0') != -1
+    - "{{ ansible_kernel | regex_replace('4.10.0-([0-9]+)-.*', '\\1') | int < 25 }}"
+
```
Comment (pavel-shirshov) [inline ansible/roles/vm_set/tasks/main.yml]: Why do we need to make this kernel upgrade? Can you please leave a comment?

File: ansible/roles/vm_set/tasks/main.yml [hunk 3]
```diff
@@ -26,11 +65,24 @@
...
+    - docker-ce
 
 - name: Install python packages
   pip: name=docker-py state=present version=1.7.2
+  become: yes
   environment: "{{ proxy_env | default({}) }}"
 
+- name: Install br_netfilter kernel module
+  become: yes
+  modprobe: name=br_netfilter state=present
```
Comment (pavel-shirshov) [inline ansible/roles/vm_set/tasks/main.yml]: Why do we need br_netfilter?

**Comment by pavel-shirshov** [comment]
Also you need to add rules for udev to make p4p1 interface. Otherwise it would have other name

**Comment by lguohan** [comment]
I'll list p4p1 as a manual step to prepare the box. I cannot find a way to include it in the playbook. I do not know which device to be p4p1.

File: ansible/ansible.cfg (hunk 1 above)
Comment (lguohan) [inline ansible/ansible.cfg]: we are really using those data. the only data we need is the kernel version and distribution, I'll decrease it to 10 minutes.

File: ansible/roles/vm_set/tasks/main.yml (hunk 3 above)
Comment (lguohan) [inline ansible/roles/vm_set/tasks/main.yml]: in roles/vm_set/test/start.yml

""
- name: Set sysctl bridge parameters for testbed
  sysctl:
    name: "{{ item }}"
    value: 0
    sysctl_set: yes
  become: yes
  with_items:
   - net.bridge.bridge-nf-call-arptables
   - net.bridge.bridge-nf-call-ip6tables
   - net.bridge.bridge-nf-call-iptables
"""

File: ansible/roles/vm_set/tasks/main.yml (hunk 2 above)
Comment (lguohan) [inline ansible/roles/vm_set/tasks/main.yml]: added comments

File: ansible/ansible.cfg (hunk 1 above)
Comment (pavel-shirshov) [inline ansible/ansible.cfg]: But ansible.cfg is global.
Our tests can use ansible facts extensively.

**Pull request merged**